<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="tripteron">
<xacro:include filename="util.xacro"/>

    <link name="world"/>
    <joint name="base" type="fixed">
        <parent link="world"/>
        <child  link="base"/>
        <origin xyz="0 0.2 0" rpy="0 0 0"/>
    </joint>
    <link name="base"/>

    <joint name="right_slider_1" type="prismatic">
        <parent link="base"/>
        <child  link="right_railAttach_one"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <axis   xyz="1 0 0"/>
        <limit effort="1000.0" lower="0.0" upper="2.0" velocity="0.5"/>
    </joint>
    
    <link name="right_railAttach_one">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0"/>	
            <geometry>
                <xacro:Cube size="${slider_size}"/>
            </geometry>
            <material name="red"/>
        </visual>
    </link>

    <joint name="right_theta_1" type="continuous">
        <parent link="right_railAttach_one"/>
        <child  link="right_lower_arm"/>
        <origin xyz="0 0 ${slider_size}" rpy="0 ${pi/4} 0"/>
        <axis   xyz="1 0 0"/>
    </joint>
    
    <link name="right_lower_arm">
        <visual>
            <origin xyz="0 0 -${toplimbLng/2}" rpy="0 0 0"/>	 
            <geometry>
                <cylinder radius="${limbRad}" length="${toplimbLng}"/> 
            </geometry>
            <material name="cyan"/>
        </visual>
    </link>
    <joint name="right_theta_2" type="continuous">
        <parent link="right_lower_arm"/>
        <child  link="right_upper_arm"/>
        <origin xyz="0 0 ${slider_size}" rpy="0 -${pi/4} 0"/>
        <axis   xyz="1 0 0"/>
    </joint>
    
    <link name="right_upper_arm">
        <visual>
            <origin xyz="0 0 -${toplimbLng/2}" rpy="0 0 0"/>	 
            <geometry>
                <cylinder radius="${limbRad}" length="${toplimbLng}"/> 
            </geometry>
            <material name="cyan"/>
        </visual>
    </link>
    
</robot>
